simulator/modules/decode/bypass/data_bypass.h

```diff
@@ -149,38 +189,38 @@ void DataBypass<ISA>::update()
         {
             if ( entry.current_stage.is_writeback())
             {
-                entry.current_stage = RegisterStage::in_RF();
+                entry.current_stage.set_to_in_RF();
                 entry.is_bypassible = false;
                 entry.is_traced = false;
             }
             else
             {
-                entry.current_stage.inc();
+                entry.change_current_stage();
 
                 if ( entry.current_stage == entry.ready_stage)
                     entry.is_bypassible = true;   
             }
         }
     }
+
+    writeback_stage_info.update();
 }
 
 
 template <typename ISA>
-void DataBypass<ISA>::untrace_instr( const Instr& instr)
+void DataBypass<ISA>::handle_flush()
 {
-    auto dst_reg_num = instr.get_dst_num();
-
-    if ( dst_reg_num.is_zero())
-        return;
-
-    if ( dst_reg_num.is_mips_hi_lo())
+    for ( auto& entry:scoreboard)
     {
-        untrace_register( Register::mips_hi );
-        untrace_register( Register::mips_lo );
-        return;
-    }    
+        if ( entry.is_traced)
+        {
+            entry.current_stage.set_to_in_RF();
```

Comment:
You may add "reset()" method to entry class and define it via default constructor

```c++
void reset() {
   *this = RegisterInfo();
}
```